tests/integration/targets/cloudfront_distribution/tasks/main.yml

```diff
@@ -126,6 +126,26 @@
       that:
         - update_origin_http_port.changed
 
+  - name: enable origin Origin Shield
+    cloudfront_distribution:
+      distribution_id: "{{ distribution_id }}"
+      origins:
+      - domain_name: "{{ cloudfront_hostname }}-origin.example.com"
+        custom_origin_config:
+          http_port: 8080
+        origin_shield:
+          enabled: true
+          origin_shield_region: '{{ aws_region }}'
+      state: present
+    register: update_origin_origin_shield
+
+  - name: ensure origin Origin Shield was enabled
+    assert:
+      that:
+        - update_origin_origin_shield.changed
+        - update_origin_origin_shield.origins.items[0].origin_shield.enabled
+        - update_origin_origin_shield.origins.items[0].origin_shield.origin_shield_region == '{{ aws_region }}'
+
```

Comment:
can you also please an idempotent test and also a 3rd one that disable it again?